exec/exploratory_plots.R

```diff
@@ -179,15 +185,65 @@ if (is.null(opt$final_assay)){
   }
 }
 
-# Guess which assays aren't logged and log them. Don't like this, but need a
-# cleaner solution
+if (is.null(opt$log2_assays)) {
+
+  print("--log2_assays param not set, will guess which input assays need to be log2.")
+  unlogged <- unlist(lapply(assay_data, function(x) max(x) > 20))
+  for (unlogged_expression_type in names(unlogged)[unlogged]) {
+    if (unlogged_expression_type %in% names(assay_data)) {
+      assay_data[[unlogged_expression_type]] <- log2(assay_data[[unlogged_expression_type]])
+    }
+  }
 
-unlogged <- unlist(lapply(assay_data, function(x) max(x) > 20))
+} else if (opt$log2_assays == "") {
+
+  print("--log2_assays param set to empty string, will not apply log2.")
+
+} else {
+  if (is_valid_positive_integer_vector(opt$log2_assays)) {
+
+    print("--log2_assays param set to list of int, will apply log2 to specified assays.")
+    unlogged <- as.integer(simpleSplit(opt$log2_assays))
+
+    # Check if log2_assays contains 0
+    if (min(unlogged) == 0) {
+
+      stop(paste("0 is not a valid assay position, use 1-based positions. Please check param --log2_assays."))
+    }
+    # Check if highest position in log2_assays is valid
+    if (max(unlogged) > length(assay_data)) {
 
-for (unlogged_expression_type in names(unlogged)[unlogged]) {
-  if (unlogged_expression_type %in% names(assay_data)) {
-    assay_data[[unlogged_expression_type]] <- log2(assay_data[[unlogged_expression_type]])
+      stop(paste(max(unlogged), "is not a valid assay position, only got", length(assay_data), "assay(s). Please check param --log2_assays."))
+    }
+    # Check for duplicates
+    if (anyDuplicated(unlogged)) {
+      stop(paste0("--log2_assays param must be unique! It contains duplicates: ", paste(unlogged[duplicated(unlogged)], collapse=", "), "."))
+    }
+
+    for (unlogged_position in unlogged) {
+      assay_data[[unlogged_position]] <- log2(assay_data[[unlogged_position]])
+    }
+  }
+
+  else {
+    print("--log2_assays param set to list of string, will apply log2 to specified assays.")
+    unlogged <- simpleSplit(opt$log2_assays)
```

Comment:
```suggestion
    unlogged <- unique(simpleSplit(opt$log2_assays))
```

Then you don't need to check for duplicates.